index.js

```diff
@@ -1,32 +1,22 @@
-// Array.prototype.find - MIT License (c) 2013 Paul Miller <http://paulmillr.com>
-// For all details and docs: https://github.com/paulmillr/array.prototype.find
-(function(globals){
-  if (Array.prototype.find) return;
+'use strict';
 
-  var find = function(predicate) {
-    var list = Object(this);
-    var length = list.length < 0 ? 0 : list.length >>> 0; // ES.ToUint32;
-    if (length === 0) return undefined;
-    if (typeof predicate !== 'function' || Object.prototype.toString.call(predicate) !== '[object Function]') {
-      throw new TypeError('Array#find: predicate must be a function');
-    }
-    var thisArg = arguments[1];
-    for (var i = 0, value; i < length; i++) {
-      value = list[i];
-      if (predicate.call(thisArg, value, i, list)) return value;
-    }
-    return undefined;
-  };
+var define = require('define-properties');
+var ES = require('es-abstract/es6');
 
-  if (Object.defineProperty) {
-    try {
-      Object.defineProperty(Array.prototype, 'find', {
-        value: find, configurable: true, enumerable: false, writable: true
-      });
-    } catch(e) {}
-  }
+var implementation = require('./implementation');
+var getPolyfill = require('./polyfill');
+var shim = require('./shim');
 
-  if (!Array.prototype.find) {
-    Array.prototype.find = find;
-  }
-})(this);
+var slice = Array.prototype.slice;
+
+var boundFindShim = function find(array, predicate) {
+	ES.RequireObjectCoercible(array);
+	return implementation.apply(array, slice.call(arguments, 1));
+};
+define(boundFindShim, {
+	implementation: implementation,
+	getPolyfill: getPolyfill,
+	shim: shim
+});
+
+module.exports = boundFindShim;
```

Comment:
I don't have trailing new lines on by default however, my editor will follow a project's style guide via [editorconfig](http://editorconfig.org/) and [eslint](http://eslint.org/). I can submit a PR with the styles you are wanting me to follow if you want?